content/blog/2020/11/2020-11-10-spring-xstream.adoc

```diff
@@ -0,0 +1,68 @@
+---
+layout: post
+title: "Spring and XStream updates (breaking changes!)"
+tags:
+- jenkins
+- core
+- developer
+author: jglick
+description: "The Spring and XStream libraries in Jenkins are being updated, including a migration to Spring Security"
+---
+
+Cleaning up technical debt is a perennial topic among Jenkins core developers,
+and one of the most visible issues is the use of obsolete and/or forked third-party libraries.
+In a world where Dependabot is offering updates to libraries released just hours before,
+it is unpleasant to be working with dependencies that are many years old.
+Since large organizations in particular are unhappy to install software using obsolete or nonstandard versions,
+my employer (CloudBees) gave its blessing for me to spend some time cleaning up some of the worst offenders.
+
+The toughest nut to crack was the Acegi Security library used for authentication,
+which has long since been replaced by Spring Security
+(and Jenkins was also bundling a long-outdated version of some Spring Framework dependencies).
+jep:227[] tracks the complicated task of updating to Spring Security
+without breaking the numerous plugins that interact with authentication,
+especially those offering a *Security Realm*.
+
+Another longstanding problem was the XStream library which Jenkins uses to read and write XML configuration files.
+This had been forked long ago by what was then the Hudson project and a few fixes applied.
+Unfortunately, some of those fixes were rejected upstream as invalid (representing unsupported usage patterns),
+and the fork fell behind the upstream version.
+jep:228[] describes the impact of switching to the upstream library in a more standard usage mode,
+including fixes to a smaller number of plugins which would otherwise be incompatible.
+
+Now that the Jenkins 2.266 weekly release includes both updates,
```

Comment:
Verify date and version number prior to merge!

Reply:
I assume that you'll be marking this as "ready for review" before the weekly release build starts Tuesday, Nov 10?

Reply:
Not sure if it needs to predate the release build but yes on Tuesday.

Reply:
Both merged toward 2.266, so just going to assume this can go out any time between now and Tuesday.